Source/API/CBLReplication.m

```diff
@@ -306,7 +306,7 @@ - (void) stop {
         [bgReplicator stop];
     }];
```

Comment:
Could you remove self as an observer right here, since you're already on the bg thread, and then not have to make another round-trip in -forgetReplication?